### PR TITLE
[docs] Incorrect use of subscription mode versus type

### PR DESCRIPTION
**Pull request by mipper**
### Motivation

concepts-messaging.md refers to SubscriptionType (Exclusive, Shared, Failover, Key_Shared) as subscription mode.; SubscriptionMode (Durable, NonDurable) is a different concept.

### Modifications

Changed mode to type where applicable.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by Anonymitaet** [comment]
Hi @mipper, "mode" is normally used to describe how messages are delivered to consumers.

image

image

image

**Comment by mipper** [comment]
Thanks for the reply.  I would classify that as a bug.  I know the documentation refers to this as mode; that was why I submitted a PR to change it to make it consistent with API.  This is, after all, developer documentation.

The Pulsar API had two concepts: SubscriptionType and SubscriptionMode.  They are completely different concepts.  The documentation refers to the concept of SubscriptionType and subscription mode which is entirely different concept in the API.  One or other should change to remove the confusion and ambiguity.  In its current form, it's completely misleading.

**Comment by Anonymitaet** [comment]
Hi @codelipenghui 

I searched `SubscriptionType` in the Pulsar repo, it refers to `Exclusive, Shared, Failover, Key_Shared`.

image

And I searched `SubscriptionMode` in the Pulsar repo, it refers to `Durable, NonDurable`.
image

In this case, shall we change `mode` to `type` for `Exclusive, Shared, Failover, Key_Shared` in docs? Thanks.
image

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -273,20 +273,20 @@ A subscription is a named configuration rule that determines how messages are de
...
+When a subscription has no consumers, its subscription type is undefined. A subscription's mode is defined when a consumer connects to the subscription, and the mode can be changed by restarting all consumers with a different configuration.
 
 ### Exclusive
 
 In *exclusive* mode, only a single consumer is allowed to attach to the subscription. If multiple consumers subscribe to a topic using the same subscription, an error occurs.
 
 In the diagram below, only **Consumer A-0** is allowed to consume messages.
 
-> Exclusive mode is the default subscription mode.
+> Exclusive mode is the default subscription type.
```
Comment (BewareMyPower) [inline site2/docs/concepts-messaging.md:289]: Should we use `Exclusive type` or just `Exclusive` to avoid misunderstanding?

**Comment by BewareMyPower** [comment]
Nice catch. I also sometimes misused the `SubscriptionMode` as the `SubscriptionType`. I also found that there's no docs about Durable & NonDurable subscription mode. This concept might be also important because `Reader`'s subscription mode is always `NonDurable`.

**Comment by jianyun8023** [comment]
I also have time to mix `subscriptionMode` and `subscriptionType`. This is a confusing place for new users who come to understand pulsar. Your findings are rigorous and great.

File: site2/docs/concepts-messaging.md (hunk 1 above)
Comment (Anonymitaet) [reply site2/docs/concepts-messaging.md:289]: I think "`Exclusive` is the default subscription type." is more accurate.

**Comment by Anonymitaet** [comment]
Thanks for your suggestions! @BewareMyPower @jianyun8023 

@mipper would you like to update docs based on our comments and add explanations for subscription mode (`durable` and `non-durable)? Thanks!

**Comment by Anonymitaet** [comment]
Hi @mipper any progress on this PR? Thanks

**Comment by sijia-w** [comment]
@BewareMyPower Shall we also correct it here? I left it unchanged at this moment.
Screen Shot 2021-10-30 at 15 14 51

@Anonymitaet More information on `subscription mode` will be added later. I still need info from eng.

**Comment by Anonymitaet** [comment]
> @Anonymitaet Sure. Will do. @BewareMyPower Regarding subscription mode, do we want to have a bit more detailed info? or [here](https://github.com/apache/pulsar/pull/10817#issuecomment-903764471) is enough?

@BewareMyPower can you give more tech inputs on `durable` and `non-durable modes? thanks

**Comment by BewareMyPower** [comment]
> > @Anonymitaet Sure. Will do. @BewareMyPower Regarding subscription mode, do we want to have a bit more detailed info? or [here](https://github.com/apache/pulsar/pull/10817#issuecomment-903764471) is enough?
> 
> @BewareMyPower can you give more tech inputs on `durable` and `non-durable` modes? thanks

I think there's no need to add related docs at this moment. The only reference of durable/non-durable subscription is [here](https://pulsar.apache.org/docs/en/pulsar-admin/#unsubscribe-1).

A non-durable subscription will be deleted when the broker stops or restarts, while a durable subscription can only be deleted by this command or the [corresponding REST API](https://pulsar.apache.org/docs/en/admin-api-topics/#unsubscribe).

File: site2/docs/concepts-messaging.md (hunk 1 above)
Comment (BewareMyPower) [reply site2/docs/concepts-messaging.md:289]: This comment is still not addressed now.

/cc @Anonymitaet

**Comment by BewareMyPower** [comment]
@sijia-w I didn't find the webpage of your screenshot. But this term is a little different from subscription mode.

- Subscription type: Exclusive, Failover, Shared, Key_Shared
  - `KeySharedMode`, the mode of Key_Shared subscription: AUTO_SPLIT, STICKY
- Subscription mode: Durable, Non-durable

**Comment by sijia-w** [comment]
@BewareMyPower Thank you for your clarification.
@Anonymitaet Actually we can find both options [here](https://pulsar.apache.org/docs/en/reference-cli-tools/#consume).

**Comment by Anonymitaet** [comment]
Fixed in https://github.com/apache/pulsar/pull/12548

**Comment by Anonymitaet** [comment]
Doc is added https://github.com/apache/pulsar/pull/12745, so close this PR.

**Pull request closed without merging**